Functions/QualityControl/xASL_qc_CollectParameters.m

```diff
@@ -34,37 +35,43 @@
     error('Missing iSubject...');
 end
 
+% It is recommended to provide session number for ASL
+if nargin<4 || isempty(iSession)
+	iSession = 1;
+	if strcmp(ScanType, 'ASL')
+		warning('Missing iSession for ASL, setting to 1');
+	end
+end
+
 if ~isfield(x,'Output')
     x.Output  = struct; 
 end
-if ~isfield(x.Output,ScanType)
+if ~isfield(x.Output, ScanType)
     x.Output.(ScanType) = struct; 
 end
 
+%% -----------------------------------------------------------------------------------------------
+%% Collect subject-specific (i.e. structural/anatomical) QC results
+
 switch ScanType
     case 'Structural'
-        QCCollectFunction = @xASL_qc_CollectQC_Structural;
+        x = xASL_qc_CollectQC_Structural(x, iSubject);
     case 'ASL'
-        QCCollectFunction = @xASL_qc_CollectQC_ASL;
+		% Check for a session subfield and create when necessary
+		if ~isfield(x.Output.ASL, x.SESSIONS{iSession})
+			x.Output.ASL.(x.SESSIONS{iSession}) = struct;
+		end
+		x = xASL_qc_CollectQC_ASL(x, iSubject, iSession);
     case 'func'
-        QCCollectFunction = @xASL_qc_CollectQC_func;
+		x = xASL_qc_CollectQC_func(x, iSubject);
```

Comment:
Why no session here? fMRI can have sessions too. Same for dwi.

Reply:
Taking all the comments together - I will add sessions also here - despite that only sessions 1 are allowed. So don't resolve yet ;)

Reply:
Done in 0b7a798c